Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+          ( ( (GetExchangeVersion -AdminDisplayVersion ($_.adminDisplayVersion.ToString())).Minor -eq 2 ) -and
+            ( (GetExchangeVersion -AdminDisplayVersion ($_.adminDisplayVersion.ToString())).Build -ge 922) ) ) } |
+        Select-Object Name, Site
+
+    if ($Sites) {
+        $uniqueSites = $null
+        $uniqueSites = $SupportedExchangeServers.Site.Name | Get-Unique
+        foreach ($site in $Sites) {
+            if ( $uniqueSites -notcontains $site ) {
+                Write-Warning "We did not find site $site"
+            }
+        }
+        $fullList = ( $SupportedExchangeServers | Where-Object { $Sites -contains $_.Site.Name } | Select-Object Name).Name
+    } else {
+        $fullList = ( $SupportedExchangeServers | Select-Object Name).Name
+    }
+
+    if ( $AllServers ) {
+        foreach ($server in $fullList) {
+            $serverName = $server.Split('.')[0]
+            if ( $SupportedExchangeServers.Name -contains $serverName ) {
+                $Version = GetWindowsMayorVersion -ExchangeServer $server
+                if ( $Version -eq 0 ) {
+                    Write-Warning "We could not get Windows version for $server."
+                    Write-Warning "Try to run the script locally."
+                } else {
+                    if ( $Version -ge 10 ) {
+                        $filterList += $serverName
+                    } else {
+                        Write-Warning "$server is not a Windows version with AMSI support."
+                    }
+                }
+            } else {
+                Write-Warning "$server is not an Exchange version with AMSI support."
+            }
+        }
+    } else {
+        foreach ($server in $ServerList) {
+            $serverName = $server.Split('.')[0]
+            if ( $fullList -contains $serverName ) {
+                if ( $SupportedExchangeServers.Name -contains $serverName ) {
+                    $Version = GetWindowsMayorVersion -ExchangeServer $server
+                    if ( $Version -eq 0 ) {
+                        Write-Warning "We could not get Windows version for $server."
+                        Write-Warning "Try to run the script locally."
+                    } else {
+                        if ( $Version -ge 10 ) {
+                            $filterList += $serverName
+                        } else {
+                            Write-Warning "$server is not a Windows version with AMSI support."
+                        }
+                    }
+                } else {
+                    Write-Warning "$server is not an Exchange version with AMSI support."
+                }
+            } else {
+                Write-Warning "We did not find any Exchange server with name: $server"
+                if ( $TestAMSI ) {
+                    $filterList += $server
+                }
+            }
+        }
+    }
+
+    if ( ( ( $filterList.count -gt 0 ) -or
```

Comment:
Nit: sometimes you use space between the brackets and sometimes you don't. I'd recommend using the same syntax everywhere in the script (I personally prefer not to use space, and this is what we do in the other scripts on CSS-Exchange too).